### DISCUSSION TITLE
Interface for v-calendar `events` prop

### DISCUSSION THREAD

**Discussion by thany** [vuetifyjs/vuetify#15870] [Q&A] [answered]
What is the interface for this prop please?

It is missing in the documentation:
![image](https://user-images.githubusercontent.com/152227/195116711-5248035f-0759-454f-9033-2f0637899b1d.png)
-> https://vuetifyjs.com/en/api/v-calendar/#props

The property `start` and `color` seem to to be parsed into something that makes sense (kind of - `color` is actually the background color, but no matter) but I have no way of knowing what other properties I can put there, or indeed what the component expects there to be.

I tried looking into the sourcecode, but it doesn't look very readable to me, with no apparant single source of prop interfaces, nor interfaces defined together with the component. It's probably perfectly readable to the author though, hence why I'm asking here.

I would appreciate if this can be added to the documentation as well. Seems pretty basic information, iyam.

We're on Vuetify 2.6 & Vue 2.6.

**Comment by KaelWD** [comment, marked as answer]
The interface is here: https://github.com/vuetifyjs/vuetify/blob/ade1434927f55a0eccf3d54f900f24c5fa85a176/packages/vuetify/types/index.d.ts#L229-L231

It's technically
```ts
interface CalendarEvent {
  start: number | string | Date
  end?: number | string | Date
  name?: string
  color?: string
}
```

but the keys for those can be modified with the `eventStart`, `eventEnd`, and `eventName` props.

> color is actually the background color

Yeah this wasn't reviewed properly when it was added, these all seem pretty inconsistent. `start`/`end`/`name` work like you'd expect, with the corresponding `event*` prop either being a key to lookup or a function called on render. `eventColor` can also be a function which is fine, but as a string it instead sets the default value for if `color` is missing from an event object. And finally `eventTextColor` can be a function, or it just sets the text color for all events with no lookup from the objects.

**Reply by thany** [reply]
Ok thanks, so I can basically only set the date and the background colour, but arbitrarily the text colour I can't set through event object, but only through a callback. Seems inconsistent to me. I will opt for setting background colour through a callback as well then, so at least in code, looking up colours looks the same.

Can this please be added to the documentation? This seems vital information, iyam.
